Sources/Commands/Workspace.swift

```diff
@@ -89,20 +92,44 @@ public class Workspace {
         /// The current version of the dependency, if known.
         public let currentVersion: Version?
 
+        /// The dependency is in editable state i.e. user is expected to modify the sources of the dependency.
+        /// The version of the dependency will not be considered during dependency resolution.
+        var isInEditableState: Bool {
+            return basedOn != nil
+        }
+
+        /// A dependency which in editable state is based on a dependency from which it edited from.
+        /// This information is useful so it can be restored when users unedit a package.
+        let basedOn: ManagedDependency?
+
         /// The current revision of the dependency.
         ///
         /// This should always be a revision corresponding to the version in the
         /// repository, but in certain circumstances it may not be the *current*
         /// one (e.g., if this data is accessed with a different version of the
         /// package manager, which would cause an alternate version to be
         /// resolved).
+        /// For a dependency in edit mode, this will be the version at which the
+        /// the dependency was made editable.
+        // FIXME: Maybe rename to lastKnownRevision?
```

Comment:
Using this for an editable dep is probably wrong. I wonder if would be better to make this optional, so clients have to deal with that (and fetch the one from the `basedOn` if they really want it). Or is it useful?